### PR TITLE
Add custom test server and additional test for conditional requests.

### LINKED ISSUE #196: Add more conditional request tests
Follow-up from #79 and #188 to add some additional test coverage.

### PR DESCRIPTION
**Pull request by netomi**
This fixes #196 .

This adds a test server that will be run using gunicorn either during the build workflow or as a separate docker image when testing locally using docker-compose.

Please take a look, if you are not happy with the solution and would like to embed it directly into the test suite I would also be fine with that ofc.

### CODE REVIEW

File: gunicorn-cfg.py
Comment (JWCook) [inline gunicorn-cfg.py:1]: Would you mind moving this config file to the `test/` directory?

**Review by JWCook — CHANGES_REQUESTED** [review]

**Comment by JWCook** [comment]
This looks good to me. Thank you very much for putting this together!

File: Dockerfile
Comment (JWCook) [inline Dockerfile:1]: Would it also be possible to move this Dockerfile to the `test/` directory?

File: gunicorn-cfg.py
Comment (netomi) [reply gunicorn-cfg.py:1]: moved  to test/server directory

**Review by JWCook — APPROVED** [review]

**Pull request merged**